### PR TITLE
Enhancement/get product

### PR DESCRIPTION
**Pull request by 5h3rr1ll**
With this PR the backend will be enhanced the ability to trigger RabbitMQ which triggers a scraper to get information about products we currently don't have in our database.

**Test Instructions**
- check out to branch enhancement/get_product
- cd into the project folder and run `npm i`
- in file docker-compose.yml change image: goodbuy/nodejs-webserver to image: goodbuy/nodejs-webserver:anthony
- run `npm run docker_dev`
- open the server and pass a product code (which we don't have in our database, just pick something from here: https://www.codecheck.info/essen/suesswaren.kat) and pass it to the URL. Schema: `localhost/product/<code>`

**Expected Behaviour**
- view is not changing
- server responding: 204, `Sending Barcode  <code>` and  `[x] Sent <code>`
- If you refresh the page after a few seconds, the product should appear on the show all product view. Alternatively, you can also look directly in the mongodb

### CODE REVIEW

**Comment by Darjusch** [comment]
Why do we have to run npm i when we use docker image with out volumes? 
Also i tried the process with the barcode 40335568 this is the [link](https://www.codecheck.info/essen/suesswaren/weitere_suesswaren/ean_40335568/id_1215047/Frigeo_Knusper_Puffreis.pro
)
Screenshot 2021-03-28 at 18 29 24
But i don't see it when i visit localhost.

File: package.json [hunk 1]
```diff
@@ -4,7 +4,7 @@
 	"description": "Node JS Api",
 	"main": "./dist/server.js",
 	"scripts": {
-		"docker_build": "docker build -t goodbuy/nodejs-webserver .",
+		"docker_build_and_push": "docker build -t goodbuy/nodejs-webserver . && docker push goodbuy/nodejs-webserver",
```
Comment (Darjusch) [inline package.json:7]: What is the reason behind the change? 
I think we first have to build the image and test it bevor we can push it. 
And if we already build it we don't have to build it again to push it. 
Maybe just leave docker_build and make an extra script docker_push?

File: dist/send_barcode.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
+#!/usr/bin/env node
+"use strict";
+let amqp = require('amqplib/callback_api');
+amqp.connect(`amqp://${process.env.RABBITMQ_USER}:${process.env.RABBITMQ_USER_PW}@${process.env.RABBITMQ_DEV_HOST}`, function (error0, connection) {
```
Comment (Darjusch) [inline dist/send_barcode.js:4]: Why is this not a ts file? 
also use strict is defined in the ts config doesn't need to be included here.

**Review by Darjusch — CHANGES_REQUESTED** [review]
One or two tests would also be nice

File: package.json (hunk 1 above)
Comment (5h3rr1ll) [reply package.json:7]: We actually talked about this change in person, but be my guest:

The "old" docker build command has no use anymore since we use to tag images tagged with our names. So we talked about to change the docker_build command, so we can use it after a branch is merged into dev. Lile this we don't forget to also push the new image to our docker repo. 

But I 'also' think it's 'dangerous' to have a script like this, caus somebody could build and push a wrong image, and then we don't know how to roll back, at least at the moment.

So I actually would say, let's remove this command completely for now and implement the build and tag step in a CI/CD.

File: dist/send_barcode.js (hunk 2 above)
Comment (5h3rr1ll) [reply dist/send_barcode.js:4]: This is just a test file that I use to test quickly if the sending works.

**Comment by 5h3rr1ll** [comment]
@Darjusch thanks for your review.

Regarding the product, it's in the mongodb as you can see here:
image

You can't see it in the frontend cause I probably forget to set also den right DB within the image I pushed, but that's not a big deal and will be fixed, as soon as we all use the same database. 😉 

When you click "changes requested" in your PR review, you should also write contritely, what changes you would like to have. 😊

File: package.json (hunk 1 above)
Comment (Darjusch) [reply package.json:7]: The CI/CD part sounds good but i like the build command i use it a lot otherwise i always have to type the command.

File: dist/send_barcode.js (hunk 2 above)
Comment (Darjusch) [reply dist/send_barcode.js:4]: Okay but we don't push js files

**Comment by Darjusch** [comment]
We agreed on writing tests for the code we write please do that.

File: dist/send_barcode.js (hunk 2 above)
Comment (5h3rr1ll) [reply dist/send_barcode.js:4]: @Darjusch we don't push js-files?
image

We sad we don't want to push js-files which are getting generated by tsc anyhow, so the PR getting cleaner. ;)

**Comment by jwdotpark** [comment]
Hey @5h3rr1ll, is this expected behavior? 
image

**Comment by 5h3rr1ll** [comment]
@Darjusch 

"We agreed on writing tests for the code we write please do that."

Help me out, what would u say should I test for this PR? 😊

**Comment by 5h3rr1ll** [comment]
@jwdotpark  ya, unfortunately, for now, this is expected. 😅

**Comment by jwdotpark** [comment]
> 
> 
> @jwdotpark ya, unfortunately, for now, this is expected. 😅

Oki no worries, i'll proceed then!

File: dist/send_barcode.js (hunk 2 above)
Comment (Darjusch) [reply dist/send_barcode.js:4]: exactly and you pushed the send_ barcode.js and send_barcode.ts

**Comment by Darjusch** [comment]
@5h3rr1ll  Calling localhost/product/<code> with a barcode. 
There you can have different cases, i would test one where the barcode isn't in the database and it successfully scrapes the product and one where it doesn't find the product.

**Review by thatsddr — APPROVED** [review]
Looks good to me!
